src/main/java/org/reaktivity/nukleus/socks/internal/stream/client/AcceptStreamProcessor.java

```diff
@@ -0,0 +1,444 @@
+            .payload(p -> p.set(
+                payload.buffer(),
+                payload.offset(),
+                payload.sizeof()))
+            .extension(e -> e.reset())
+            .build();
+        correlation.connectEndpoint()
+            .accept(
+                dataForwardFW.typeId(),
+                dataForwardFW.buffer(),
+                dataForwardFW.offset(),
+                dataForwardFW.sizeof());
+    }
+
+    private void handleConnectThrottleAfterHandshake(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+        case WindowFW.TYPE_ID:
+            final WindowFW window = context.windowRO.wrap(buffer, index, index + length);
+            handleWindow(window);
+            break;
+        case ResetFW.TYPE_ID:
+            final ResetFW reset = context.resetRO.wrap(buffer, index, index + length);
+            handleReset(reset);
+            break;
+        default:
+            // ignore
+            break;
+        }
+    }
+
+    private void handleConnectThrottleBeforeHandshake(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch (msgTypeId)
+        {
+        case WindowFW.TYPE_ID:
+            final WindowFW window = context.windowRO.wrap(buffer, index, index + length);
+            connectWindowBytes += window.update();
+            connectWindowFrames += window.frames();
+            correlation.nextAcceptSignal().accept(isConnectReplyStateReady);
+            break;
+        case ResetFW.TYPE_ID:
+            final ResetFW reset = context.resetRO.wrap(buffer, index, index + length);
+            handleReset(reset);
+            break;
+        default:
+            // ignore
+            break;
+        }
+    }
+
+    private void handleWindow(
+        WindowFW window)
+    {
+        final int targetWindowBytesDelta = window.update();
+        final int targetWindowFramesDelta = window.frames();
+
+        final int sourceWindowBytesDelta = targetWindowBytesDelta + sourceWindowBytesAdjustment;
+        final int sourceWindowFramesDelta = targetWindowFramesDelta + sourceWindowFramesAdjustment;
+
+        sourceWindowBytesAdjustment = Math.min(sourceWindowBytesDelta, 0);
+        sourceWindowFramesAdjustment = Math.min(sourceWindowFramesDelta, 0);
+
+        if (sourceWindowBytesDelta > 0 || sourceWindowFramesDelta > 0)
+        {
+            doWindow(correlation.acceptThrottle(), correlation.acceptStreamId(), Math.max(sourceWindowBytesDelta, 0),
+                Math.max(sourceWindowFramesDelta, 0));
+        }
+    }
+
+    private void handleReset(
+        ResetFW reset)
+    {
+        doReset(correlation.acceptThrottle(), correlation.acceptStreamId());
+    }
+
+    @Signal
+    public void attemptNegotiationRequest(boolean isConnectReplyBeginFrameReceived)
+    {
+        if (dataNegotiationRequestFW == null)
```

Comment:
When is this ever != null and isn't that a bug if it is wrapped again by another stream?

Reply:
if it is always null then why not eliminate the reference and always get it from the context?

Reply:
A new stream would be handled by a new instance of AcceptStreamProcessor, this is my understanding. It would be not null if the previous time there was not a large enough window available for the data to be.

Reply:
But the context object is created in the ClientStreamFactory, which is shared across streams.  So if you are reusing a stared FW object across calls to process() on a stream, then it is possible a different stream called wrap, so the data would be corrupted.  

I'm not fully following what is going on here with the Signal() and who calls this so maybe it is not the case that the FW is shared across calls to process?  (i.e. this method is called twice in one event loop/not over multiple frames arriving)